app/state_machines/flood_risk_engine/enrollment_state_machine.rb

```diff
@@ -14,8 +14,54 @@ class EnrollmentStateMachine < FiniteMachine::Definition
     initial :check_location
 
     events do
-      event :go_forward, WorkFlow.for(:local_authority)
-      event :go_back, WorkFlow.for(:local_authority).invert
+      event :go_forward, WorkFlow.for(:start).merge(
+        if: -> { target.org_type.nil? }
+      )
+      event :go_back, WorkFlow.for(:start).invert.merge(
+        if: -> { target.org_type.nil? }
+      )
+
+      event :go_forward, WorkFlow.for(:local_authority).merge(
+        if: -> { target.org_type == "local_authority" }
+      )
+      event :go_back, WorkFlow.for(:local_authority).invert.merge(
+        if: -> { target.org_type == "local_authority" }
+      )
+
+      event :go_forward, WorkFlow.for(:limited_company).merge(
+        if: -> { target.org_type == "limited_company" }
+      )
+      event :go_back, WorkFlow.for(:limited_company).invert.merge(
+        if: -> { target.org_type == "limited_company" }
+      )
+
+      event :go_forward, WorkFlow.for(:limited_liability_partnership).merge(
```

Comment:
I've tried a number of ways of doing just that, but can't get any of them to work. The problem is that the `target` object is defined on the fly by the underlying state engine, and the whole script is defined within the code of the state engine's configuration. That means that you can just add loops or methods that would tidy this up.